### PR TITLE
Add support for prettier links through angle brackets

### PR DESCRIPTION
**Pull request by LumosX**
Based on #5 (so the feature improves upon the fixed version in that PR; don't merge before #5). Replaces the default URL parsing for nameless and named internal links and lets them use the `[](<#some target>)` syntax instead of getting URLs encoded. Thus, a URL containing a space (or, presumably, other special characters) will be parsed as `[](<#target tiddler>)` instead of `[target tiddler](#target%20tiddler)`, which is much more preferable if you ask me.

For example, the test tiddler
> ```This is a test tiddler that features an [[internal link]] followed by another [[different link]] and then by a [[third one|named link]], and then one more named link here [[like this|another named link]].```

... becomes

> ```This is a test tiddler that features an [](<#internal link>) followed by another [](<#different link>) and then by a [third one](<#named link>), and then one more named link here [like this](<#another named link>).```

This can be suppressed by running the script with the `--no-angle-brackets` flag, which will produce the previously-standard output:

> ```This is a test tiddler that features an [](#internal%20link) followed by another [](#different%20link) and then by a [third one](#named%20link), and then one more named link here [like this](#another%20named%20link).```

... which parses links like URLs.

Why have I set the new option to be the default? Because parsing internal links as URLs seems to be overkill, the %20s are ugly as all hell, and I'll likely want to extract the markdown info into a different format further down the line, and the last thing I'd want to deal with are going to be the silly URL encodings. Feel free to modify this as you wish.

### CODE REVIEW

**Comment by LumosX** [comment]
Updated and rebased this against the latest version of the branch for PR 5. I also couldn't stop myself from making the parse_link function more elegant.

**Comment by MaxGyver83** [comment]
> Updated and rebased this against the latest version of the branch for PR 5. I also couldn't stop myself from making the parse_link function more elegant.

Sorry, I have squashed your commits of PR #5. I think this caused a conflict. You probably can resolve this by rebasing and dropping the first 5 commits of this PR.

**Comment by LumosX** [comment]
Turns out some dialects of markdown, unlike tiddlywiki's, don't like links with empty square brackets. This includes github too, so I'm guessing tiddlywiki is the outlier, and a simple link like `[](#nowhere)` ends up looking like: [](#nowhere) <- this blank space.

This means that anyone who actually wants to export the markdown for later use will be exceptionally annoyed if my angle-bracket option is the default. As a result, I've inverted the flag so it's optional (and fixed a typo in the help description, oops)

> Sorry, I have squashed your commits of PR #5. I think this caused a conflict. You probably can resolve this by rebasing and dropping the first 5 commits of this PR.

No worries, can do while I'm here.

Oh, and the last commit is a correction of a typo where a caught exception was being named "error" instead of "err".

**Comment by MaxGyver83** [comment]
> Turns out some dialects of markdown, unlike tiddlywiki's, don't like links with empty square brackets. This includes github too, so I'm guessing tiddlywiki is the outlier, and a simple link like `[](#nowhere)` ends up looking like: <- this blank space.

True, I noticed the same when I created a static wiki using `md2html` ([md4c](https://github.com/mity/md4c)).

File: tid2md.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def write_markdown_file(lines: list, md_file: Path) -> bool:
                 # External links
                 # [[text|url]] → [text](url)
                 # [[url]] → <url>
-                line = re_link.sub(parse_link, line)
+                line = re_link.sub(lambda m: parse_link(m, use_angles), line)
```
Comment (MaxGyver83) [inline tid2md.py:196]: I guess you turned this into a lambda (again) by accident while rebasing!? Or does it bring any advantage over `re_link.sub(parse_link(m, use_angles), line)`

Comment (LumosX) [reply tid2md.py:196]: This was intentional. You're throwing me for a loop here (again, been some years since I used Python), so I had to double-check. Running `re_link.sub(parse_link(m, use_angles), line)` in the python REPL produces a NameError as I expected, because "m" is not a defined variable; running the script itself throws an "'NoneType' object has no attribute 'group'" error, because "m" is never set to anything. The delegate function that re.sub is waiting for must take one argument only, it seems.

The only thing I can come up with to avoid wrapping parse_link in a lambda is to make it curryable, e.g.
```
def parse_link(use_angles: bool) -> str:
    def parse_inner(match: re.Match):
        # Processing unchanged, returns a string
    return parse_inner
```
This is kind of exactly the same as wrapping it in a lambda though, but it allows us to use `line = re_link.sub(parse_link(use_angles), line)`. I'm quite certain this isn't what you had in mind though... or was it?

In short, I'm not exactly sure what you mean, your proposed alternative doesn't seem to work. Could you elaborate? (Don't tell me it's my windows again, I'm running anaconda like a sane person. :P)

Comment (MaxGyver83) [reply tid2md.py:196]: OK, now I get it. I was even wondering why the first argument of `sub` is a function at all (instead of a string). Now I see that this first argument is like the second argument in `re.sub` which can be either a string or a function. I didn't even notice that your were passing the `parse_link` function before (and not a function call).

**Review by MaxGyver83 — APPROVED** [review]
